### ISSUE TITLE
allowuninstall doesn't work properly with multilib

### ISSUE THREAD

**Issue by ignatenkobrain** [openSUSE/libsolv#187]
```
repo system 0 testtags <inline>
#>=Pkg: lz4 1 1 x86_64
#>=Prv: liblz4.so.1()(64bit)
#>=Pkg: lz4 1 1 i686
#>=Prv: liblz4.so.1()
#>=Pkg: zzz 1 1 i686
#>=Req: liblz4.so.1()
repo available 0 testtags <inline>
#>=Pkg: lz4 1 1 x86_64
#>=Prv: liblz4.so.1()(64bit)
#>=Pkg: lz4 1 1 i686
#>=Prv: liblz4.so.1()
#>=Pkg: lz4 2 1 x86_64
#>=Pkg: lz4-libs 2 1 x86_64
#>=Prv: liblz4.so.1()(64bit)
#>=Pkg: lz4-libs 2 1 i686
#>=Prv: liblz4.so.1()
system x86_64 rpm system
poolflags implicitobsoleteusescolors
solverflags allowvendorchange keepexplicitobsoletes bestobeypolicy keeporphans yumobsoletes
job update all packages [forcebest]
job allowuninstall pkg lz4-1-1.x86_64@system
job allowuninstall pkg lz4-1-1.i686@system
result transaction,problems <inline>
#>upgrade lz4-1-1.x86_64@system lz4-2-1.x86_64@available
#>upgrade lz4-1-1.i686@system lz4-libs-2-1.i686@available
#>install lz4-libs-2-1.x86_64@available
```

```
+problem f80ba3dc info cannot install both lz4-2-1.x86_64 and lz4-1-1.x86_64
+problem f80ba3dc solution 87c9006c allow lz4-1-1.i686@system
+problem f80ba3dc solution 998c80be allow lz4-1-1.x86_64@system
```

But that's not true, because you don't need lz4.i686, you need only something which provides `liblz4.so.1()(64bit)`.

**Comment by mlschroe** [comment]
That's not really an allowunistall problem, but a problem with the 'forcebest' option.

**Comment by ignatenkobrain** [comment]
Without forcebest it doesn't upgrade anything btw.

**Comment by mlschroe** [comment]
This should be fixed with commit 22623468dcdfeeaafeec3bd6789dfaf739eb7664

**Comment by ignatenkobrain** [comment]
@mlschroe looks like if I remove `forcebest` now, it will not update anything. But this is wrong, no?
